### DISCUSSION TITLE
토큰을 담은 Cookie에 대해서

### DISCUSSION THREAD

**Discussion by kjklalala2** [heading1/WYLSBingsu#195] [Q&A]
저희가 저번 시연 영상에서     
로그인으로 이동 안하고 빙수로 이동한 것에 대해서 쿠키에 토큰이 남아있어서 이러한 현상이 생겼었습니다.    
근데 저희가 Refresh/Access Token 정책을 사용하다 보니 

- 로그아웃 상황을 
- Refresh/Access 토큰 둘 다 만료되었을 때

이 때 빼고는 토큰을 삭제하거나 유효 시간을 설정하는 것이 안될 것 같습니다.    

쿠키에 AccessToken이 있는데 유효 시간을 설정해서 날려버리는 것은 만료되었는지 판단이 불가능하고,    
 Access 토큰이 만료되었어도 RefreshToken을 통해 다시 발급할 것인지 혹은 다시 로그인을 할 것인지 판별하는 것이기 때문입니다. 

백엔드 측면에서 생각해보았고 프론트는 잘 모르곘어요. 의견 부탁드려요!

**Comment by Circlewee** [comment]
일단 프론트에서 직접 토큰을 건들일 수가 없습니다. 왜냐하면 저번에 설정한 httpOnly 옵션때문에요.
이부분은 제가 할수 있는 얘기는 딱히 없는 것 같고 백엔드 분들이 해결해주셔야 할것 같아요ㅠ

**Reply by kjklalala2** [reply]
저희 로그아웃은 따로 API가 없는 것 같은데 로그아웃 API 만들어서 쿠키를 삭제하는 방법은 어떻게 생각하세요?    
Refresh/Access Token 둘 다 만료 되었을 때는 미들웨어에서 쿠키를 삭제하면 되는데 의견 부탁드려요!

**Reply by Circlewee** [reply]
그렇다면 로그아웃을 하지 않은 사용자의 토큰 만료는 어떻게 알수 있나요??
로그아웃은 만들어 보겠습니다!
